src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,110 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+export const NewMovie = ({
+  onAdd = () => {},
+}: {
+  onAdd?: (movie: Movie) => void;
+}) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [currentTitle, setCurrentTitle] = useState('');
+  const [currentDescription, setCurrentDescription] = useState('');
+  const [currentImdbUrl, setCurrentImdbUrl] = useState('');
+  const [currentImdbId, setCurrentImdbId] = useState('');
+  const [currentImgUrl, setCurrentImgUrl] = useState('');
+
+  const allFieldsAreValid = () => {
+    const urlFields = ['imdbUrl', 'imgUrl'];
+    const urlPattern = new RegExp(
+      `^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\\w]+@)?[A-Za-z0-9.-]+|` +
+        `(?:www\\.|[-;:&=+$,\\w]+@)[A-Za-z0-9.-]+)` +
+        `((?:\/[+~%/\\.\\w-_]*)?\\??(?:[-+=&;%@,\\w_]*)#?(?:[,.!/\\\\\\w]*))?)$`,
+    );
+
+    const urlFieldsValid = urlFields.every(field => {
+      const fieldValue = field === 'imdbUrl' ? currentImdbUrl : currentImgUrl;
+
+      return urlPattern.test(fieldValue);
+    });
+
+    return currentTitle && urlFieldsValid && currentImdbId;
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (allFieldsAreValid()) {
+      onAdd({
+        title: currentTitle.trim(),
+        description: currentDescription.trim(),
+        imgUrl: currentImgUrl.trim(),
+        imdbUrl: currentImdbUrl.trim(),
+        imdbId: currentImdbId.trim(),
+      });
+
+      setCurrentTitle('');
+      setCurrentDescription('');
+      setCurrentImdbUrl('');
+      setCurrentImdbId('');
+      setCurrentImgUrl('');
+
+      setCount(count + 1);
+    }
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form className="NewMovie" key={count} onSubmit={handleSubmit}>
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={currentTitle}
+        onChange={title => setCurrentTitle(title)}
         required
       />
 
-      <TextField name="description" label="Description" value="" />
+      <TextField
+        name="description"
+        label="Description"
+        value={currentDescription}
+        onChange={description => setCurrentDescription(description)}
```

Comment:
```suggestion
        onChange={setCurrentDescription}
```
it can be shorter. same for other inputs